[16.0][MIG] base_export_manager: Migration to version 16.0

stefan-tecnativa opened the pull request:
cc @Tecnativa TT43220

This Pr superseeds: #649 

@CarlosRoca13 @chienandalu please review!

pedrobaeza committed:
[REN] base_exports_manager > base_export_manager

hhgabelgaard committed:
Missing rename

yajo committed:
[IMP] base_export_manager: Several things (II):

* Improve UX for base_export_manager.

  - Improve user instructions in README.
  - Require some required fields.
  - Allow to select models from a list.
  - Allow to select up to 3 fields from dynamic lists.
  - Improve translations.
  - More tests.
  - Translate column labels.

  Some methods have been renamed, so version tag is raised to 8.0.2.0.0.

* Hardcode constraints in the `create` method instead of using normal ones.

  Depending on the context where the record is created, it gets `resource` or `model_id`. The problem is that Odoo checks constrains before inverses, so constrains would fail always.

  Test added to ensure future versions contemplate all use cases.

* Allow to reorder fields.

yajo committed:
[IMP] base_export_manager: Several things:

* Improve UX for base_export_manager.

  - Improve user instructions in README.
  - Require some required fields.
  - Allow to select models from a list.
  - Allow to select up to 3 fields from dynamic lists.
  - Improve translations.
  - More tests.
  - Translate column labels.

  Some methods have been renamed, so version tag is raised to 8.0.2.0.0.

* Make inverse method be called at view time.
* Hardcode constraints in the `create` method instead of using normal ones.

  Depending on the context where the record is created, it gets `resource` or `model_id`. The problem is that Odoo checks constrains before inverses, so constrains would fail always.

  Test added to ensure future versions contemplate all use cases.

* Allow to reorder fields.

oca-transbot committed:
OCA Transbot updated translations from Transifex

lasley committed:
[MIG] base_export_manager: Upgrade to v9
* Upgrade base_export_manager to v9
* Rename JS to base_export_manager

lasley committed:
base_export_manager: Upgrade to v9 * Upgrade base_export_manager to v9 * Rename JS to base_export_manager

lasley committed:
[REF] base_export_manager: Fix model domain
* Change osv_memory to transient for model_id domain

Commit:
Add export permission

Commit:
[IMP] Add export permission

Commit:
IMP: support v7,v8 API and remove sql with ORM methods (#1)

Commit:
[FIX+IMP] PEP8 + conflicts + Combined methods

yajo committed:
[8.0][FIX][base_export_manager] Fix "Expected singleton" bug. (#521)

[FIX][base_export_manager] Fix "Expected singleton" bug.

If you had a field that got translated in more than 1 addon, you'd possibly getto this error:

      File "/opt/odoo/0079_ahk_openerp/oca/base_export_manager/models/ir_exports_line.py", line 105, in _compute_label
        field.name)),
      File "/opt/odoo/common/openerp/v8/openerp/fields.py", line 825, in __get__
        record.ensure_one()
      File "/opt/odoo/common/openerp/v8/openerp/models.py", line 5355, in ensure_one
        raise except_orm("ValueError", "Expected singleton: %s" % self)
    except_orm: ('ValueError', 'Expected singleton: ir.translation(4899, 703976)')

With this patch, now we let Odoo return the translated string by using its
standard method to do so, so we have to care for less.

* Move installation outside a data file.

This makes the whole installation to be able to roll back if something goes
wrong, instead of entering an error loop.

* Include envorionment in its manager.

* Add 4th field

* Move to api.multi, refactoring some stuff.

- Add some comments in complex parts.
- Rename `onchange_name` to `_onchange_name` (guidelines).
- Make `_compute_name`'s try block shorter and easier to understand.

* Allow R/W of name directly in model.

* Update tests to cover new behaviors.

Commit:
[MIG]Migrated base_export_manager module.

yajo committed:
[MIG][base_export_manager] Migrate features from 8.0

yajo committed:
[MIG] base_export_manager: Migrate to v11

- Remove all possible translation commits
- Aesthetical changes
- Bump version
- Some new standards, such as split readme

Basically, nothing.

hhgabelgaard committed:
Translated using Weblate (Danish)

Currently translated at 80.0% (20 of 25 strings)

Translation: server-ux-11.0/server-ux-11.0-base_export_manager
Translate-URL: https://translation.odoo-community.org/projects/server-ux-11-0/server-ux-11-0-base_export_manager/da/

Commit:
[MIG] base_export_manager: Migration to 12.0

pedrocs-exo committed:
Translated using Weblate (Portuguese)

Currently translated at 100.0% (31 of 31 strings)

Translation: server-ux-12.0/server-ux-12.0-base_export_manager
Translate-URL: https://translation.odoo-community.org/projects/server-ux-12-0/server-ux-12-0-base_export_manager/pt/

liweijie0812 committed:
Translated using Weblate (Chinese (Simplified))

Currently translated at 100.0% (31 of 31 strings)

Translation: server-ux-12.0/server-ux-12.0-base_export_manager
Translate-URL: https://translation.odoo-community.org/projects/server-ux-12-0/server-ux-12-0-base_export_manager/zh_CN/

badbole committed:
Translated using Weblate (Croatian)

Currently translated at 90.3% (28 of 31 strings)

Translation: server-ux-12.0/server-ux-12.0-base_export_manager
Translate-URL: https://translation.odoo-community.org/projects/server-ux-12-0/server-ux-12-0-base_export_manager/hr/

Tardo committed:
[IMP] base_export_manager: black, isort, prettier

Tardo committed:
[MIG] base_export_manager: Migration to 13.0

kevinkhao committed:
[IMP] base_export_manager: black, isort, prettier

kevinkhao committed:
[MIG] base_export_manager: Migration to 14.0

anasuarez1 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (31 of 31 strings)

Translation: server-ux-14.0/server-ux-14.0-base_export_manager
Translate-URL: https://translation.odoo-community.org/projects/server-ux-14-0/server-ux-14-0-base_export_manager/es/

pedrobaeza committed:
[FIX] base_export_manager: Don't return on invalid field, but break

If not, we are not assigning the label value for the record and we get
a missing cache value error.

nicomacr committed:
[IMP] base_export_manager: pre-commit execution

nicomacr committed:
[MIG] base_export_manager: Migration to 15.0

victoralmau committed:
[FIX] base_export_manager: Remove default because it is redundant

WARNING odoo.fields: Redundant default on ir.exports.line.model1_id

syera94 committed:
[MIG] base_export_manager : Migration to 16.0

rjpathan committed:
[IMP] base_export_manager: pre-commit stuff

rjpathan committed:
[MIG] base_export_manager: Migration to 16.0

chienandalu commented on base_export_manager/models/ir_exports_line.py:
```diff
@@ -142,7 +142,8 @@ def _inverse_name(self):
                 # invalidate_cache -> in order to get actual value of field 'label'
                 # in function '_check_name'
-                self.invalidate_cache(ids=one.ids)
+                # self.env["ir.exports.line"].invalidate_cache(ids=one.ids)
```
Don't leave commented code
```suggestion
```

chienandalu commented on base_export_manager/models/ir_exports_line.py:
```diff
@@ -142,7 +142,8 @@ def _inverse_name(self):
             if any(parts):
                 # invalidate_cache -> in order to get actual value of field 'label'
```
Update the comment

chienandalu reviewed (changes requested):
Functionally looks good to me. Some minor code comments:

(also, some commits can be squashed)

pedrobaeza commented:
Did you see #563 ?

chienandalu commented:
This superseeds a more complete PR. Anyway, it has a better approach to the invalidate_recordset approach is maybe more clear and self explanatory

pedrobaeza commented:
Then it's fair to credit the author of the first PR ever. What about rebasing this one over the other?

chienandalu commented:
Ok, it shouldn't be that hard to cherry-pick and resolve a pair of conflicts.

stefan-tecnativa commented:
> Ok, it shouldn't be that hard to cherry-pick and resolve a pair of conflicts.

I see the situation, what I do then? Rebase this PR with the other one? Or I cherry-pick the migration commit of the other PR, resolve the conflicts and, eventually, squash this PR's migration commit maintaining the authorship?

pedrobaeza commented:
Save your current work, cherry pick the migration commit of the other PR, and put your relevant changes on it, doing another commit.

chienandalu commented:
Well, cherry pick both migration commits indeed, as you've got two contributors to keep credit to

stefan-tecnativa commented:
@chienandalu it's the history okay now? I will now review your comments

chienandalu commented on base_export_manager/views/ir_exports.xml:
```diff
@@ -29,52 +29,48 @@
         <field name="model">ir.exports</field>
         <field name="arch" type="xml">
             <form string="Export Profile">
-                <sheet>
```
You're dismissing the changes of the previous commit here.

stefan-tecnativa replied:
Thanks for the advice, now it should be correct

stefan-tecnativa committed:
[FIX] base_export_manager: ListController corrected file

kikopeiro reviewed (approved):
LGTM :+1 (without code review)

Thanks!

pedrobaeza commented:
/ocabot migration base_export_manager

pedrobaeza committed:
[FIX] base_export_manager: Avoid access error when using export profile

Steps to reproduce:

- Create an export profile in any model.
- Login with a user without setting permissions.
- Go to export popup on that model.
- Try to select the saved export profile.

You'll get an access error.

Using sudo for computed/related stuff linked to ir.model,  we avoid
the problem.

TT44721

pedrobaeza commented:
Please include #704

chienandalu reviewed (approved):
:+1:

pedrobaeza commented:
/ocabot merge nobump
